File: docs/scripting/samples/python-scripts/python-examples/Event_Display_Protocol_Properties.py
"""
Event Display Protocol Properties

Python scripting example demonstrating Cellario API integration and automation capabilities
"""

#Cellario modules
from HRB.Cellario.Scripting import *
from HRB.Cellario.Scripting.API import *


def Execute(api : PythonScriptingApi):
    #about this script 
    #info about this script from api.CurrentRun() and api.CurrentProtocol()
    runorderId = api.CurrentPlate.CurrentRun.RunNumber
    runOrderDesc = api.CurrentPlate.CurrentRun.Description
    protocolName = api.CurrentPlate.CurrentProtocol.ProtocolName
    
    message = "Before Order State Change for run = " + str(runorderId) + " \n"
    message += "Order Description '" + runOrderDesc + "' \n"
    message += "Order State - " + str(api.CurrentPlate.CurrentRun.RunState) + " \n"
    message += "Protocol Name - " + protocolName + " \n"
    api.Messaging.Notify("Before Order State Change", message)
    
    
    properties = api.CurrentPlate.CurrentProtocol.Properties
    
    index = 0
    for property in properties:
        index += 1
        pname = property.PropertyName
        pid = property.PropertyId
        pvalue = property.PropertyValue
        pdesc = property.PropertyDescription
    
        message = "Property " + str(index) + " Name is '" + pname + "' \n"
        message += "Property Id is '" + str(pid) + "' \n"
        message += "Property Value is '" + pvalue + "' \n"
        message += "Property Desc is '" + pdesc + "' \n"
    
        api.Messaging.Notify("Protocol Property " + str(pname), str(message))

File: docs/scripting/samples/csharp-scripts/examples/Event_Display_Protocol_Properties.cs
/**
 * Event Display Protocol Properties
 * 
 * Example script showing basic Cellario protocol automation and customization features
 */

#region Header
// ** Copyright HighRes Biosolutions, Inc. 2020 **
// 
//    File:      EventScript_DisplayProtocolProperties.cs 
//    Project:   ScriptLibrary
//    Solution:  CellarioWPF
#endregion

using System;
using System.Linq;
using HRB.Cellario.Scripting.API;

namespace HRB.Cellario.Scripting.ScriptLibrary
{
    public class EventScript_DisplayProtocolProperties :AbstractScript
    {
        /// <summary>
        /// Method called during Cellario protocol execution when a sample arrives at the scripting step.
        /// </summary>
        /// <remarks>Executes synchronously with the run scheduler. Device operation results not available
        /// until <see cref="ReleaseResources"/> method is called.</remarks>
        /// <param name="api">Access to properties and methods for interacting with the Cellario run-time scheduler, 
        /// samples, resources and devices operations.</param>
        public override void Execute(IScriptingApi api)
        {
            //about this script 
            //info about this script from api.CurrentRun() and api.CurrentProtocol()
            var runorderId = api.CurrentRun.RunNumber;
            var runOrderDesc = api.CurrentRun.Description;
            var protocolName = api.CurrentRun.Protocol.ProtocolName;

            var message = "Before Order State Change for run = " + runorderId.ToString() + " \n";
            message += "Order Description '" + runOrderDesc + "' \n";
            message += "Order State - " + api.CurrentRun.RunState + " \n";
            message += "Protocol Name - " + protocolName + " \n";
            api.Messaging.Notify("Before Order State Change", message);

            var properties = api.CurrentRun.Protocol.Properties.ToList();

            var index = 0;
            foreach (var property in properties)
            {
                index++;
                var pname = property.PropertyName;
                var pid = property.PropertyId;
                var pvalue = property.PropertyValue;
                var pdesc = property.PropertyDescription;

                message = "Property " + index.ToString() + " Name is '" + pname + "' \n";
                message += "Property Id is '" + pid + "' \n";
                message += "Property Value is '" + pvalue + "' \n";
                message += "Property Desc is '" + pdesc + "' \n";

                api.Messaging.Notify("Protocol Propert " + index.ToString(), message);
            }
        }
    }
}
